facades/PC/build.gradle.kts

```diff
@@ -30,396 +28,370 @@ dateTimeFormat.timeZone = TimeZone.getTimeZone("UTC")
+    return when {
+        Os.isFamily(Os.FAMILY_WINDOWS) -> "windows"
+        Os.isFamily(Os.FAMILY_MAC) -> "macosx"
+        Os.isFamily(Os.FAMILY_UNIX) -> "linux"
+        else -> {
+            logger.warn("What kind of libraries do you use on this? {}", System.getProperty("os.name"))
+            "UNKNOWN"
+        }
     }
 }
 
 
-ext {
-    // Default path to store server data if running headless via Gradle
-    localServerDataPath = 'terasology-server'
+// Default path to store server data if running headless via Gradle
+val localServerDataPath by extra("terasology-server")
 
-    // General props
-    mainClassName = 'org.terasology.engine.Terasology'
-    subDirLibs = 'libs'
-    templatesDir = new File(rootDir, 'templates')
-    rootDirDist = new File(rootDir, 'build/distributions')
+// General props
+val mainClassName by extra("org.terasology.engine.Terasology")
+val subDirLibs = "libs"
+val templatesDir = File(rootDir, "templates")
+val rootDirDist = File(rootDir, "build/distributions")
 
-    // Read environment variables, including variables passed by jenkins continuous integration server
-    env = System.getenv()
+// Inherited props
+val dirNatives: String by rootProject.extra
+val distsDirectory: DirectoryProperty by project;
+
+// Read environment variables, including variables passed by jenkins continuous integration server
+val env: MutableMap<String, String> = System.getenv()!!
+
+// Version related
+val startDateTimeString = dateTimeFormat.format(Date())!!
+val versionFileName = "VERSION"
+val versionBase by lazy { File(templatesDir, "version.txt").readText().trim() }
+val displayVersion = versionBase
 
-    // Version related
-    startDateTimeString = dateTimeFormat.format(new Date())
-    versionFileName = 'VERSION'
-    versionBase = new File(templatesDir, "version.txt").text.trim()
-    displayVersion = versionBase
-}
 
 application {
-    mainClass = project.ext.mainClassName
+    mainClass.set(extra.get("mainClassName") as String)
 }
 
 // Adjust as the Gradle 6 upgrade changed this path a bit
 sourceSets {
-    main.java.outputDir = new File("$buildDir/classes")
-    test.java.outputDir = new File("$buildDir/testClasses")
+    main { java.outputDir = File("$buildDir/classes") }
+    test { java.outputDir = File("$buildDir/testClasses") }
 }
 
 // Base the engine tests on the same version number as the engine
-version = project(':engine').version
-println "PC VERSION: $version"
+version = project(":engine").version
+logger.info("PC VERSION: {}", version)
 
 // Jenkins-Artifactory integration catches on to this as part of the Maven-type descriptor
-group = 'org.terasology.facades'
+group = "org.terasology.facades"
 
 dependencies {
-    implementation project(':engine')
-    implementation group: 'org.reflections', name: 'reflections', version: '0.9.10'
+    implementation(project(":engine"))
+    implementation(group = "org.reflections", name = "reflections", version = "0.9.10")
```

Comment:
On a side note: https://github.com/ronmamo/reflections is available in 0.9.12 now.

Reply:
It has been for a while, but there are some [widely reported issues](https://github.com/ronmamo/reflections/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) with 0.9.12 (and some for 0.9.11) so I've been afraid to mess with it.